src/jitlayers.cpp

```diff
@@ -1629,6 +1629,12 @@ JuliaOJIT::JuliaOJIT()
     DLSymOpt(std::make_unique<DLSymOptimizer>(false)),
     ContextPool([](){
         auto ctx = std::make_unique<LLVMContext>();
+        if (!ctx->hasSetOpaquePointersValue())
```

Comment:
```suggestion
```
this will be always true, and is removed in the next LLVM version

Reply:
Not true, when you run with `JULIA_LLVM_ARGS="-opaque-pointers"` to force Julia to use them, e.g., for testing purposes, that flag is `false`. Ignoring it and calling `setOpaquePointers` results in an assertion failure.

Reply:
Ah, I see it also calls that in the constructor if the flag exists
```
2362c4ecdc881 llvm/lib/IR/LLVMContextImpl.cpp     (Arthur Eubanks          2022-03-21 17:11:35 -0700  50)   if (OpaquePointersCL.getNumOccurrences()) {
2362c4ecdc881 llvm/lib/IR/LLVMContextImpl.cpp     (Arthur Eubanks          2022-03-21 17:11:35 -0700  51)     OpaquePointers = OpaquePointersCL;
2362c4ecdc881 llvm/lib/IR/LLVMContextImpl.cpp     (Arthur Eubanks          2022-03-21 17:11:35 -0700  52)   }
```

Reply:
Anyways, git seems to indicate that hasSetOpaquePointersValue only exists in 15.x, not 14.x (which didn't have the assertion for setOpaquePointers, only for enableOpaquePointers) or 16.x (which doesn't support typed pointers)